ferry.go

```diff
@@ -292,6 +292,102 @@ func (f *Ferry) Initialize() (err error) {
 	return nil
 }
 
+func (f *Ferry) CatchUpOnMissedBinlogsOnResume() error {
+	if f.StateToResumeFrom == nil {
+		return nil
+	}
+
+	tableSchemaCache := f.StateToResumeFrom.LastKnownTableSchemaCache
+
+	// Step 1: setup the binlog reconciler
+	binlogReconciler := &BinlogReconciler{
+		BatchSize:        f.Config.BinlogEventBatchSize,
+		TableSchemaCache: tableSchemaCache,
+
+		SourceDB:     f.SourceDB,
+		BinlogWriter: f.BinlogWriter,
+	}
+	binlogReconciler.Initialize()
+
+	// TODO: Step 2: setup the iterative verifier reconciler
+
+	// Step 3: initialize the binlog streamer and the throttler
+	binlogStreamer := f.NewBinlogStreamer()
```

Comment:
wouldn't it be cleaner to have this inside the reconciler?

Reply:
The BinlogReconciler is supposed to catch up binlogs events that occured while Ghostferry is down. However, if we want to interrupt/resume the verifier, the process of resuming that will also need an instance of the binlog streamer. Since we don't want two binlog streamers, this is managed directly in the Ferry.